SPECK/src/main/java/SPEck.java

```diff
@@ -227,19 +296,22 @@ protected static void writeDataset(String fileOut, Utils.Dataset dataset) throws
     /**
      * Compute the p-values of the FSP of a dataset merging the p-values computed by the different cores.
      * @param   T               the number of random datasets for the Monte Carlo estimate of p-values
-     * @param   parallelization the number of cores of the machine. It is the number of random datasets
+     * @param   numCores        the number of cores of the machine. It is the number of random datasets
      *                          generated at the same time in the Monte Carlo estimation. Each core of
-     *                          the machine computes the p-values on T/parallelization random datasets.
+     *                          the machine computes the p-values on T/numCores random datasets.
      * @param   fileMined       the file containing the FSP extracted from the dataset stored in file
      * @return
      */
-    private int[] computePValues(int T, int parallelization, String fileMined){
+    private int[] computePValues(int T, 
+                                 int numCores, 
+                                 String fileMined)
+    {
         // the indexes of the cores of the machine
         IntArrayList indexes = new IntArrayList();
-        for (int i = 0; i < parallelization; i++) indexes.add(i+1);
+        for (int i = 0; i < numCores; i++) indexes.add(i+1);
         // computes in parallel the p-values from different random datasets and merges the p-values computed by the different cores
```

Comment:
computes in parallel the number of datasets in each batch with results more or as extreme as the one in `fileFSP`, and then aggregates the results from each batch to obtain the complete number of the T datasets.